lib/matplotlib/backend_bases.py

```diff
@@ -3057,21 +3113,21 @@ def press_zoom(self, event):
         if (event.button not in [MouseButton.LEFT, MouseButton.RIGHT]
                 or event.x is None or event.y is None):
             return
-        axes = [a for a in self.canvas.figure.get_axes()
-                if a.in_axes(event) and a.get_navigate() and a.can_zoom()]
-        if not axes:
-            return
-        if self._nav_stack() is None:
-            self.push_current()  # set the home button to this view
+
+        axes = self._start_event_axes_pan(event)
```

Comment:
Do we need an `if not axes: return` here so we don't add the id_zoom below unnecessarily?

Reply:
yes that makes sense... implemented!